src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -186,12 +186,11 @@ static DiagnosticInfo createObsoleteDiagnostic(Symbol symbol, BinderFlags locati
 
                 if (data.Kind == ObsoleteAttributeKind.Experimental)
                 {
-                    Debug.Assert(data.Message is null);
                     Debug.Assert(!data.IsError);
 
                     // Provide an explicit format for fully-qualified type names.
                     return new CustomObsoleteDiagnosticInfo(MessageProvider.Instance, (int)ErrorCode.WRN_Experimental, data,
-                        new FormattedSymbol(symbol, SymbolDisplayFormat.CSharpErrorMessageFormat));
+                        new FormattedSymbol(symbol, SymbolDisplayFormat.CSharpErrorMessageFormat), string.IsNullOrEmpty(data.Message) ? "" : " - ", data.Message ?? "");
```

Comment:
Why deviate from existing design, ie. relying on pairs of resource strings (`WRN_DeprecatedSymbolStr` corresponding to `WRN_DeprecatedSymbol` for example)? That feels safer and more flexible from a localization point of view.

Reply:
I am not following. I do not think this change violates any design and I do not think it affects flexibility or localization in any noticeable way.

Reply:
Based on offline discussion, I am going to make a change